pkg/api/authz.go

```diff
@@ -161,33 +157,28 @@ func (ac *AccessController) getUserGroups(username string) []string {
 	return groupNames
 }
 
-// getContext updates an AccessControlContext for a user/anonymous and returns a context.Context containing it.
-func (ac *AccessController) getContext(acCtx *localCtx.AccessControlContext, request *http.Request) context.Context {
-	readGlobPatterns := ac.getGlobPatterns(acCtx.Username, acCtx.Groups, Read)
-	dmcGlobPatterns := ac.getGlobPatterns(acCtx.Username, acCtx.Groups, DetectManifestCollision)
+// getContext updates an UserAccessControl with admin status and specific permissions on repos.
+func (ac *AccessController) updateUserAccessControl(userAc *reqCtx.UserAccessControl) {
+	readGlobPatterns := ac.getGlobPatterns(userAc.GetUsername(), userAc.GetGroups(), Read)
```

Comment:
I think all types of patterns and actions should be available to the NewUserAccessControl() in case we need other types of permissions per extensions.